include/ghex/rma/uuid.hpp

```diff
@@ -0,0 +1,59 @@
+/* 
+ * GridTools
+ * 
+ * Copyright (c) 2014-2020, ETH Zurich
+ * All rights reserved.
+ * 
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * 
+ */
+#ifndef INCLUDED_GHEX_RMA_UUID_HPP
+#define INCLUDED_GHEX_RMA_UUID_HPP
+
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <boost/lexical_cast.hpp>
+#include <string>
+
+namespace gridtools {
+namespace ghex {
+namespace rma {
+
+/** @brief Universally unique identifier used for labelling shared memory regions. */
+struct uuid
+{
+    boost::uuids::uuid m_id;
+    std::string m_name;
+
+    struct info
+    {
+        char m_name[39];
+    };
+
+    uuid()
+    : m_id{ boost::uuids::random_generator{}() }
+    , m_name{ "s-" + boost::lexical_cast<std::string>(m_id)}
+    { }
+
+    uuid(const uuid&) = default;
+    uuid(uuid&&) = default;
+
+    const boost::uuids::uuid& id() const noexcept { return m_id; }
+    const std::string& name() const noexcept { return m_name; }
+
+    info get_info() const noexcept
+    {
+        info info_;
+        for (int i=0; i<38; ++i) info_.m_name[i] = m_name[i];
```

Comment:
Just double checking since I'm not familiar with uuids in boost: is the 38 the right value? should it be some variable length?

Reply:
from the docs:

> The string representation of a uuid is hhhhhhhh-hhhh-hhhh-hhhh-hhhhhhhhhhhh where h is hexadecimal.

which are 36 characters. With the prefix and the '\0' at the end, we'll get 39 chars.